packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkConfigurationExt.kt

```diff
@@ -75,30 +76,101 @@ internal fun ReadableMap.asProxyConfig(): Pair<Proxy, ProxyAuthenticator?>? {
         port = getInt("port")
     }
 
-    val type =
-        getString("type")?.let {
-            when (it.lowercase(Locale.US)) {
+    val type = getString("type")
+
+    return if (address != null && port != null && type != null) {
+        JSONProxyConfiguration(
+            type,
+            address,
+            port,
+            getString("username"),
+            getString("password")
+        ).asProxyConfig()
```

Comment:
The idea was to avoid to duplicate the logic, but I can rather extract the logic in another function that I call in both methods. I think this will make more sense since it won't couple the 2 interfaces together 👍